src/hypothesis/DrPerceptron.py

```diff
@@ -15,23 +15,28 @@ def __init__(self, windowSize=100):
         self._threshold = 0.5
         self._learning_rate = 0.1
 
-        self._weightsSize = 5
-        self._weights = [0] * self._weightsSize
 
-        self._inputWindow =  [0] * self._weightsSize
+        """ Note: _inputList is the list of inputs and the historical
+                  window of inputs on which to train _weights.  Current implementation
+                  is a sliding window in a modded universe defined by self._windowSize
+        """
+        self._windowSize = 5
```

Comment:
Basically it's inputs + weights and no nested layers.  _windowSize governs all of this.